opentelemetry/src/trace/context.rs

```diff
@@ -185,6 +185,26 @@ impl SpanRef<'_> {
         self.with_inner_mut(move |inner| inner.update_name(new_name))
     }
 
+    /// Adds a [`Link`] to another [`SpanContext`].
```

Comment:
while you are at it, could you also add a note stating
1. Links added via this mechanism is not accessible for Samplers.
2. It is recommended to add Links at Span creation time, rather than adding it after Span creation.

Relevant spec part:
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#link